### ISSUE TITLE
Windows OS open folders don't get focus while Windows Terminal installed.

### ISSUE THREAD

**Issue by willcampbell7** [microsoft/terminal#9644]
### Windows Terminal version (or Windows build number)

1.6.10571.0

### Other Software

Windows OS 10.0.19042.868

### Steps to reproduce

Install Windows Terminal.

### Expected Behavior

When selecting folder from Task Bar jump list, the folder opens and gets focus.
When selecting folder in Start Menu, the folder opens, gets focus, and the Start Menu closes.

### Actual Behavior

It is not necessary that Windows Terminal be running. Whatever happens is because it is installed, not because it is running, and it infects the Windows system. In the Task Bar, right-click File Explorer to select a folder pinned (folder appearing in the 'jump list'). Folder opens but icon in Task Bar flashes because focus has not shifted to the newly opened folder. Click folder to get focus. Eventually worse symptoms develop. Most noticeable the Start Menu doesn't close when selecting a folder (Documents, Downloads, etc.), The folder opened using the Start Menu doesn't get focus, the Start Menu stays open, and again the icon in Task Bar flashes to alert user the folder needs attention. Uninstall Windows Terminal and all focus problems are gone. Reinstall Windows Terminal and problems return. There is a discussion in the Microsoft Community regarding this issue. Many users have confirmed the problem and that Windows Terminal is the cause. https://answers.microsoft.com/en-us/windows/forum/windows_10-start-win_menu/start-menu-not-closing-after-opening-anything-from/eaae4553-a746-47f8-ae6a-acd62bbf6187

**Comment by miniksa** [comment]
The only thing I can think of here is that somehow the Open Terminal Here shell extension is upsetting `explorer.exe` and causing this weird behavior. We will have to look into it.

**Comment by renard11** [comment]
Thanks willcampbell7 for the bug report! I've been having this issue for a couple of months and it drives me mad! I never would have suspected Windows Terminal to be the reason for taht. I hope there'll be a fix soon - I removed Windows Terminal from all of my machines for the time being, and it's so refreshing not to struggle with folders opening in the background!

I just want to add that jumplists have always worked fine for me, it was just the search dialog and folders pinned to the start menu.

**Comment by DHowett** [comment]
So, you'll both be surprised to learn that we already know about this one! Weirdly enough, it seems like a Windows bug that's related to us having (_not even using! just **having** it will cause the bug!_) a registration for a shell extension (the "Open Windows Terminal here") link.

Unfortunately, since it's a Windows bug, there's not much we can do about it on the Terminal team. Either we remove a very highly-trafficked and well-loved feature, or we live with a moderate annoyance :smile:

More info in the original bug, #7122. I'll make sure that we have an internal bug for the team who owns this, as well.

/dup #7122 (sorry to make you chase a line of duplicates!)

**Comment by renard11** [comment]
> So, you'll both be surprised to learn that we already know about this one! Weirdly enough, it seems like a Windows bug that's related to us having (_not even using! just **having** it will cause the bug!_) a registration for a shell extension (the "Open Windows Terminal here" menu item).
> 
> Unfortunately, since it's a Windows bug, there's not much we can do about it on the Terminal team. Either we remove a very highly-trafficked and well-loved feature, or we live with a moderate annoyance 😄
> 
> More info in the original bug, #7122. I'll make sure that we have an internal bug for the team who owns this, as well.
> 
> /dup #7122 (sorry to make you chase a line of duplicates!)

You call this a "moderate annoyance"??? This must be a joke! Moreover, the "Open Windows Terminal here" menu doesn't even work for me when I use WSL2! It works in WSL1, though. When I use the feature in WSL2, I always end up in my home folder, no matter in what folder I press "open terminal here", so it's absolutely useless for me. But searching for folders is essential to me...

**Comment by renard11** [comment]
> So, you'll both be surprised to learn that we already know about this one! Weirdly enough, it seems like a Windows bug that's related to us having (_not even using! just **having** it will cause the bug!_) a registration for a shell extension (the "Open Windows Terminal here" menu item).
> 
> Unfortunately, since it's a Windows bug, there's not much we can do about it on the Terminal team. Either we remove a very highly-trafficked and well-loved feature, or we live with a moderate annoyance 
> 
> More info in the original bug, #7122. I'll make sure that we have an internal bug for the team who owns this, as well.
> 
> /dup #7122 (sorry to make you chase a line of duplicates!)

I have to admit, I'm speechless! I have to wrap my mind around this - so you've known about this problem since July 2020. A problem that is somehow caused by the "Open Windows Terminal here" extension which breaks a core functuality of Windows, and you think you should still keep this feature and put this app into the Microsoft Store for people to download and therefore unknowingly break their system?!?! Because you feel like it's just, as you call it, a "moderate annoyance", which is somehow made up for by the well-loved "open windows terminal here" feature?

Have you any idea how many people are annoyed by this, that have Windows Terminal installed but have no idea that it's this app that's causing that problem, and would gladly remove it in order to be able to properly conduct a simple search if they had any knowledge about it?

Me and my workgroup have to deal with hundreds of folders on a daily basis, and what you call a moderate annoyance becomes a not-so-moderate annoyance pretty fast. We have spent hours with clueless Microsoft support agents, that were stuck in a feedback loop between "have you tried updating your system" and "have you tried a fresh install", all the while a simple right-click on Windows Terminal and "uninstall" would have sufficed in order to get our workflow back, which we did yesterday. Ubuntu in WSL runs perfectly fine without the admittedly much better looking Windows Terminal...

**Comment by willcampbell7** [comment]
Thank you for your response. I looked at the existing issue and yes it sounds the same. I see a workaround there that I will try tomorrow. I'll gladly give up the the context menu option if it solves the other problems. I hardly use that feature anyway, and if I might suggest, like many other programs, the user should have to the option to disable it anyway. An entry in the settings JSON would be perfect.

I suspected this problem came about in combination with a Windows update, because I'd been using Windows Terminal for some time without this problem, then the problem started a few months ago. Also, in the Microsoft Community thread there are users who don't use Windows Terminal, it's not installed, but say this happens in connection with using OneDrive. Perhaps it too has a context menu addition screwing up something the Windows team has overlooked in a recent update.

I want to be constructive here and offer any help making this issue less painful for us all. I can post more details at the other issue or here. What is the appropriate thing to do? By the way, before posting a new issue, I searched for a similar issue and nothing matched, even after myriad combination of phrases and keywords that should have easily matched. In fact, I can type into the search box the exact title of this thread -- result is no matches found. Someone might want to look into that.

Anyway, I'm not here to complain. I want to help solve issues. Anything I can do, let me know. Thanks. William
